linux/src/main/cpp/DarkModeGnome.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Jannis Weis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "com_github_weisj_darkmode_platform_linux_gnome_GnomeNative.h"
+
+#include <string>
+#include <glibmm-2.4/glibmm.h>
+#include <giomm-2.4/giomm.h>
+
+JNIEXPORT jstring JNICALL
+Java_com_github_weisj_darkmode_platform_linux_gnome_GnomeNative_getCurrentTheme(JNIEnv *env, jclass) {
+  Gio::init();
+  Glib::RefPtr<Gio::Settings> settings = Gio::Settings::create("org.gnome.desktop.interface");
+  return env->NewStringUTF(settings->get_string("gtk-theme").c_str());
+}
+
+struct EventHandler {
+  std::string themeSettingName = "gtk-theme";
+  JavaVM *jvm;
+  JNIEnv *env;
+  jobject callback;
+  Glib::RefPtr<Gio::Settings> settings;
+  sigc::connection settingsChangedSignalConnection;
+
+  void settingChanged(const Glib::ustring &name) {
+    runCallBack();
+  }
+
+  void runCallBack() {
+    bool detachNecessary = false;
+    int getEnvStat = jvm->GetEnv((void **) &env, JNI_VERSION_1_6);
+    if (getEnvStat == JNI_EDETACHED) {
+      detachNecessary = true;
+      if (jvm->AttachCurrentThread((void **) &env, NULL) != 0) return;
+    } else if (getEnvStat == JNI_EVERSION) {
+      return;
+    }
+
+    jclass runnableClass = env->GetObjectClass(callback);
+    jmethodID runMethodId = env->GetMethodID(runnableClass, "run", "()V");
+    if (runMethodId) {
+      env->CallVoidMethod(callback, runMethodId);
+    }
+    if (env->ExceptionCheck()) {
+      env->ExceptionDescribe();
+    }
+    if (detachNecessary)
+      jvm->DetachCurrentThread();
+  }
+
+  void stop() {
+    settingsChangedSignalConnection.disconnect();
```

Comment:
I'm not sure whether it is necessary but it may be the best to manually delete `settingsChangedSignalConnection` in the destructor of `EventHandler`. (Better be safe than sorry)

Reply:
`Cannot delete expression of type 'sigc::connection'`
If [this SO answer](https://stackoverflow.com/a/18940502/1687436) is to be believed, `settingsChangedSignalConnection` can't be deleted as it is not a pointer created by `EventHandler`.

Reply:
Good to know. Then this isn't needed.